### PR TITLE
docs: add 'creating a fuel dapp' section

### LINKED ISSUE #1913: Docs: Creating a Fuel dApp
Transform the current Quickstart into a tutorial to build a simple app.

We must show how to leverage our tooling and use the `create fuels` CLI  to scaffold a new full-stack dApp. We also need to go over the generated code and explain what everything is doing.

We should also think of an alternate step-by-step to achieve the same as the scaffolded app.

For example:
 1. Write a sample contract
 2. Build your contract
 3. Running local node
 4. Deploying contract
 5. Generating types
 6. Calling contracts
    -  👉 _In the last section, we can have a cross-link that ties it back with the rest of the docs_

### PR DESCRIPTION
**Pull request by Dhaiwat10**
## Summary

This PR adds:

- A new page under _Introduction_ > _Creating a Fuel dApp_
- Moves the old _Quickstart_ page containing docs related to `npm create fuels` to _Tooling_ -> _CLI_ -> `npm create fuels`

https://github.com/FuelLabs/fuels-ts/assets/39617427/c09b5724-3f51-418f-a4ff-9d814a125afd

Closes #1913

### CODE REVIEW

File: apps/docs/.vitepress/config.ts
Comment (petertonysmith94) [inline apps/docs/.vitepress/config.ts:55]:
```suggestion
            text: 'Creating a Fuel dApp',
            link: '/guide/introduction/creating-a-fuel-dapp',
```

I believe moving this to the introduction folder will pass the validation.
To pass the validation from [here](https://github.com/FuelLabs/github-actions/blob/master/docs-hub/vp-docs.js).

File: apps/docs/src/guide/creating-a-fuel-dapp/index.md
Comment (petertonysmith94) [inline apps/docs/src/guide/creating-a-fuel-dapp/index.md:97]:
```suggestion
## Dev Environment Setup
```

File: templates/nextjs/src/pages/index.tsx
Comment (petertonysmith94) [inline templates/nextjs/src/pages/index.tsx:71]:
```suggestion
        <Link href="https://docs.fuel.network/docs/fuels-ts/fuels/">
```

File: apps/create-fuels-counter-guide/README.md
Comment (petertonysmith94) [inline apps/create-fuels-counter-guide/README.md:22]:
```suggestion
- [Fuel TS SDK docs](https://docs.fuel.network/docs/fuels-ts/)
```

File: apps/create-fuels-counter-guide/README.md
Comment (petertonysmith94) [inline apps/create-fuels-counter-guide/README.md:1]: Did you mean to reference the repository for `create-fuels`?

File: apps/create-fuels-counter-guide/src/pages/index.tsx
Comment (petertonysmith94) [inline apps/create-fuels-counter-guide/src/pages/index.tsx:89]:
```suggestion
        This template uses the new
```

File: apps/create-fuels-counter-guide/src/pages/index.tsx
Comment (petertonysmith94) [inline apps/create-fuels-counter-guide/src/pages/index.tsx:90]:
```suggestion
        <Link href="https://docs.fuel.network/docs/fuels-ts/fuels/">
```

File: apps/create-fuels-counter-guide/src/pages/index.tsx
Comment (petertonysmith94) [inline apps/create-fuels-counter-guide/src/pages/index.tsx:92]:
```suggestion
        </Link>
```

Now i'm thinking there is a reason behind this?

File: apps/docs/src/guide/creating-a-fuel-dapp/index.md
Comment (petertonysmith94) [inline apps/docs/src/guide/creating-a-fuel-dapp/index.md:3]:
```suggestion
`create fuels` is a command line tool that helps you scaffold a new fullstack Fuel dApp. In this guide, we will create a new counter dApp using `create fuels` and add decrement functionality to it. The final result will look like this:
```

Is the package manager relevant in this case?

Comment (Dhaiwat10) [reply apps/docs/src/guide/creating-a-fuel-dapp/index.md:3]: I can't say yes or no. I just want the _brand_ of this package to be `npm create fuels` because the name itself is the command. If we only keep it `create-fuels` or `create fuels` then that's gonna be separate from the command, if you know what I mean?

File: apps/create-fuels-counter-guide/src/pages/index.tsx
Comment (Dhaiwat10) [reply apps/create-fuels-counter-guide/src/pages/index.tsx:92]: Probably the formatter or eslint

File: apps/create-fuels-counter-guide/src/pages/index.tsx
Comment (Dhaiwat10) [reply apps/create-fuels-counter-guide/src/pages/index.tsx:90]: I'll have to make these changes in the `create-fuels` template since this project is just bootstrapped from there. Mind if I file a different issue for these changes?

File: apps/create-fuels-counter-guide/README.md
Comment (Dhaiwat10) [reply apps/create-fuels-counter-guide/README.md:1]: Same as https://github.com/FuelLabs/fuels-ts/pull/1965#discussion_r1549213016

File: apps/create-fuels-counter-guide/README.md
Comment (Dhaiwat10) [reply apps/create-fuels-counter-guide/README.md:22]: Same as https://github.com/FuelLabs/fuels-ts/pull/1965#discussion_r1549213016

File: templates/nextjs/src/pages/index.tsx
Comment (Dhaiwat10) [reply templates/nextjs/src/pages/index.tsx:71]: Same as https://github.com/FuelLabs/fuels-ts/pull/1965#discussion_r1549213016

File: apps/docs/.vitepress/config.ts
Comment (Dhaiwat10) [reply apps/docs/.vitepress/config.ts]: Why so? It seems to be complaining about the file name for some reason. Not where the file is placed

File: apps/docs/src/guide/creating-a-fuel-dapp/index.md
Comment (petertonysmith94) [reply apps/docs/src/guide/creating-a-fuel-dapp/index.md:3]: That does indeed make sense - it's all about the brand!

File: apps/create-fuels-counter-guide/src/pages/index.tsx
Comment (petertonysmith94) [reply apps/create-fuels-counter-guide/src/pages/index.tsx:90]: For sure @Dhaiwat10

File: apps/docs/.vitepress/config.ts
Comment (petertonysmith94) [reply apps/docs/.vitepress/config.ts]: The issue lies [here](https://github.com/FuelLabs/github-actions/blob/master/docs-hub/vp-docs.js#L134C1-L135C1), where we're not replacing all the spaces for hyphens (for a folder name).

image

---

By moving the doc, we can bypass this logic and trigger a functional [replacement](https://github.com/FuelLabs/github-actions/blob/master/docs-hub/vp-docs.js#L167).

(This is for sure a hack, but thought I'd share the knowledge)

Comment (Dhaiwat10) [reply apps/docs/.vitepress/config.ts]: Oh that's annoying. Thanks for pointing this out and sharing the solution!

File: apps/docs/src/guide/introduction/creating-a-fuel-dapp.md
Comment (danielbate) [inline apps/docs/src/guide/introduction/creating-a-fuel-dapp.md:42]:
```suggestion
⚡️ Success! Created a fullstack Fuel dApp at my-fuel-project
```

File: apps/docs/src/guide/introduction/creating-a-fuel-dapp.md
Comment (danielbate) [inline apps/docs/src/guide/introduction/creating-a-fuel-dapp.md:129]:
```suggestion
You should now be able to see the counter dApp running at `http://localhost:4000`. You can try changing the contents of the `./sway-programs/contract/src/main.sw` file and see the changes reflected in the UI without having to restart the server.
```

File: apps/docs/src/guide/introduction/creating-a-fuel-dapp.md
Comment (danielbate) [inline apps/docs/src/guide/introduction/creating-a-fuel-dapp.md:149]: IMO this should be split into FE and BE - implement the program then the dApp.

File: apps/docs/src/guide/introduction/creating-a-fuel-dapp.md
Comment (danielbate) [inline apps/docs/src/guide/introduction/creating-a-fuel-dapp.md:167]:
```suggestion
Congratulations! That's all. You should now be able to see the counter dApp running at `http://localhost:4000` with our newly added decrement functionality.
```

File: apps/docs/src/guide/introduction/creating-a-fuel-dapp.md
Comment (danielbate) [inline apps/docs/src/guide/introduction/creating-a-fuel-dapp.md:170]: I think a link here to the source code to show where to find the finished code would be good. Or should this go at the start?

File: apps/create-fuels-counter-guide/fuels.config.ts
Comment (Dhaiwat10) [reply apps/create-fuels-counter-guide/fuels.config.ts:9]: I am not sure _why_, but I have tried using them and it didn't work. cc @arboleya

File: apps/docs/src/guide/cli/npm-create-fuels.md
Comment (Dhaiwat10) [reply apps/docs/src/guide/cli/npm-create-fuels.md:1]: Yeah it will break the link. Probably best kept the way it is.

File: apps/docs/src/guide/introduction/creating-a-fuel-dapp.md
Comment (Dhaiwat10) [reply apps/docs/src/guide/introduction/creating-a-fuel-dapp.md:25]: We should, but this is not a command. This is the output of the command.

File: apps/docs/src/guide/introduction/creating-a-fuel-dapp.md
Comment (Dhaiwat10) [reply apps/docs/src/guide/introduction/creating-a-fuel-dapp.md:42]: I'm afraid I'll have to make this change in the CLI tool's code in a separate PR when I update those docs links because I have just copied over the exact message that the tool logs out when you run it.

File: apps/docs/src/guide/introduction/creating-a-fuel-dapp.md
Comment (Dhaiwat10) [reply apps/docs/src/guide/introduction/creating-a-fuel-dapp.md:129]: `3000` is actually correct, since that's the port where the frontend runs.

File: apps/docs/src/guide/introduction/creating-a-fuel-dapp.md
Comment (Dhaiwat10) [reply apps/docs/src/guide/introduction/creating-a-fuel-dapp.md:167]: Same as above

File: apps/create-fuels-counter-guide/README.md
Comment (Dhaiwat10) [reply apps/create-fuels-counter-guide/README.md:18]: You're correct, but the same as https://github.com/FuelLabs/fuels-ts/pull/1965#discussion_r1553258633

File: apps/docs/src/guide/introduction/creating-a-fuel-dapp.md
Comment (Dhaiwat10) [reply apps/docs/src/guide/introduction/creating-a-fuel-dapp.md]: Great feedback. I added it here: https://github.com/FuelLabs/fuels-ts/pull/1965/commits/448ca705f41354dda10d436cc2ba37579c3e3b29

File: apps/docs/src/guide/introduction/creating-a-fuel-dapp.md
Comment (Dhaiwat10) [reply apps/docs/src/guide/introduction/creating-a-fuel-dapp.md:174]: So this would have to be a link to the `create-fuels-counter-guide` app on the `fuels-ts` repo, but since we haven't merged this PR to `master` yet, that link doesn't exist yet. Do you mind if I add it via another small PR once the changes have been merged?

**Comment by Dhaiwat10** [comment]
@danielbate re. your comment about deployment instructions, we are still working on making the template deployment ready and figuring out what would be the best strategy for it. Expect something very soon on this. I will make sure I add it to our docs.

File: apps/docs/src/guide/introduction/creating-a-fuel-dapp.md
Comment (Dhaiwat10) [reply apps/docs/src/guide/introduction/creating-a-fuel-dapp.md:42]: #2016

File: apps/docs/src/guide/introduction/creating-a-fuel-dapp.md
Comment (danielbate) [reply apps/docs/src/guide/introduction/creating-a-fuel-dapp.md:129]: Ah my b, 4000 was the node ofc.

File: apps/docs/src/guide/introduction/creating-a-fuel-dapp.md
Comment (danielbate) [reply apps/docs/src/guide/introduction/creating-a-fuel-dapp.md:174]: Ah ofc!

File: apps/create-fuels-counter-guide/fuels.config.ts
Comment (danielbate) [reply apps/create-fuels-counter-guide/fuels.config.ts:9]: This is just desirable, should not block the PR, feel free to resolve.

**Review by danielbate — APPROVED** [review]

**Pull request merged**
